pysages/backends/hoomd.py

```diff
@@ -129,4 +135,25 @@ def bind(context, sampling_method, **kwargs):
     sampler = Sampler(method_bundle, sync_and_bias)
     context.integrator.cpp_integrator.setHalfStepHook(sampler)
     #
+    SAMPLERS_CONTEXTS.append((sampler, context))
```

Comment:
if we make this a dictionary, with the context as key and sampler as value, much of the logic for detach would be simpler, right?

Reply:
we just have to have a logic to make sure that every context has only one sampler, or we need a list of samplers as values

Reply:
Good point. This can be handled in bind I think. As HOOMD only accepts one `HalfStepHook` at a time.

Reply:
Wait. This is fine, `setHalfStepHook` replaces previously bound samplers. And since we are using a dict now, reassigning the entry will also remove the reference to any previous sampler.